runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowMetrics.java

```diff
@@ -350,10 +352,18 @@ public static MetricQueryResults create(
     public abstract MetricName name();
     public abstract String step();
     @Nullable
-    public abstract T committed();
-    @Nullable
+    protected abstract T committedInternal();
     public abstract T attempted();
 
+    public T committed() {
+      T committed = committedInternal();
+      if (committed == null) {
+        throw new UnsupportedOperationException("This runner does not currently support committed"
```

Comment:
I'm not so keen on the use of exception instead of null. If something is explicitly optional `@Nullable` seems reasonable (insert standard rant about how `Optional` is a better choice but Java screwed it up and Guava is shaded).

I would imagine that changes to this class would just remove `@Nullable` from attempted, then an additional static factory method like `createCommitted(<just committed value>)` that populates both fields with that value, `createAttempted(<just attempted value>)` that leaves committed null and `create(<both not nullable>)`. Incidentally the existing `create` method seems broken as it fails to mark them as `@Nullable`. When we plug in the checker framework that should be caught.

Reply:
I see - the javadoc tells you to do this. That sucks. Want to fix the javadoc to have this better spec?

Reply:
OK let's just make Dataflow match the spec and later change the spec to be good.